context/csv2js.py

```diff
@@ -70,52 +77,93 @@ def read_csvs(csvs, primary_key):
     }
 
 
-def get_column(column, list_of_dicts):
+def get_raw_column(column, list_of_dicts):
+    '''
+    >>> get_raw_column('a', [{'a':'1'}])
+    ['1']
+    '''
+    includes_nones = [d.get(column) for d in list_of_dicts]
+    return [val for val in includes_nones if val is not None]
+
+
+def get_typed_column(column, list_of_dicts):
     '''
-    >>> get_column('a', [{'a':1}, {'b':2}])
-    [1, None]
+    >>> get_typed_column('a', [{'a':'1'}])
+    [1]
+    >>> get_typed_column('a', [{'a':'1.1'}])
+    [1.1]
     '''
-    return [d.get(column) for d in list_of_dicts]
+    return [typed(s) for s in get_raw_column(column, list_of_dicts)]
+    # TODO: could this be a generator?
 
 
-def all_numbers(l):
+def all_type(l, t):
     '''
-    >>> all_numbers(['1','one'])
+    >>> all_type([1, 'one'], int)
     False
-    >>> all_numbers(['-1.1e-1'])
+    >>> all_type([-1.1e-1], float)
     True
-    >>> all_numbers([])
+    >>> all_type([], int)
     True
     '''
-    return all([is_number(x) for x in l if x is not None])
+    return all(type(x) == t for x in l)
 
 
-def is_number(s):
+def typed(s):
     '''
-    >>> is_number('1')
-    True
-    >>> is_number('z')
-    False
+    >>> typed('1.1')
+    1.1
+    >>> typed('0')
+    0
+    >>> typed('z')
+    'z'
     '''
-    if s is None:
-        return None
+    assert type(s) in [str]
```

Comment:
maybe the `in` is overkill for a one element list? `type(s) == str`

Reply:
yep. Thanks.